mri/operators/fourier/cartesian.py

```diff
@@ -88,18 +95,26 @@ def op(self, img):
             images the coils dimension is put first
         """
         if self.n_coils == 1:
-            return self._mask * np.fft.ifftshift(np.fft.fftn(
+            return self.mask * np.fft.ifftshift(np.fft.fftn(
                                     np.fft.fftshift(img), norm="ortho"))
         else:
             if self.n_coils > 1 and self.n_coils != img.shape[0]:
                 raise ValueError("The number of coils parameter is not equal"
                                  "to the actual number of coils, the data must"
                                  "be reshaped as [n_coils, Nx, Ny, Nz]")
             else:
-                # TODO: Use joblib for parallelization
-                return np.asarray([self._mask * np.fft.ifftshift(np.fft.fftn(
-                                    np.fft.fftshift(img[ch]), norm="ortho"))
-                                   for ch in range(self.n_coils)])
+                axes = tuple(np.arange(1, img.ndim))
+                return self.mask * np.fft.ifftshift(
+                    np.fft.fftn(
```

Comment:
Did you check (and if so how) that `fftn` is indeed faster than the loop?

Indeed I had this problem, summarized in this [SO question](https://stackoverflow.com/questions/58009758/numpy-fftn-very-inefficient-for-2d-fft-of-several-images) a long time ago.

Finally, it might be nice to consider some parallelization on this (maybe for now as a TODO).

Reply:
Well, there was a TODO for parallelization, but then I assumed that the fftn through numpy might as well handle this well and  in practice this should not be that big a gain, but you do have a valid point. I will check this and get back here.

Reply:
moved to scipy

Reply:
Also added n_jobs and connected the argument to workers. This way, user has control on exact speed